imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,207 @@
+"""Class for creating and managing CDF attrs.
+
+Developed based of HermesDataSchema from HERMES-SOC/hermes_core.
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import yaml
+
+DEFAULT_GLOBAL_CDF_ATTRS_FILE = "imap_default_global_cdf_attrs.yaml"
+DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE = "default_global_cdf_attrs_schema.yaml"
+DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE = "default_variable_cdf_attrs_schema.yaml"
+
+
+class CdfAttributeManager:
+    """
+    Class for creating and managing CDF attributes based out of yaml files.
+
+    This class is based on the HERMES SWxSOC project for managing CDF attributes, but
+    is intended to be a flexible and very lightweight way of managing CDF attribute
+    creation and validation.
+
+    To use, you can load one or many global and variable attribute files:
```

Comment:
You might want to add a sentence on which way the update works for loading. Feel free to modify my suggestion or move it elsewhere.
```suggestion
    To use, you can load one or many global and variable attribute files
    (the latter files will overwrite duplicate entries in the previous ones):
```